packages/react/src/avatar-context/AvatarContext.tsx

```diff
@@ -0,0 +1,13 @@
+import { type ComponentPropsWithoutRef, createContext } from "react";
+
+import type { Avatar } from "../avatar";
+
+type AvatarContextType = {
+  size: ComponentPropsWithoutRef<typeof Avatar>["size"];
+};
+
+export const AvatarContext = createContext<{
+  size: AvatarContextType["size"] | undefined;
```

Comment:
we can inline the size - no need for an extra `AvatarContextType` type
```suggestion
export const AvatarContext = createContext<{
  size: ComponentPropsWithRef<typeof Avatar>["size"] | undefined;
```